### PR TITLE
Update Get-DbaFile.ps1

### PR DESCRIPTION
**Pull request by 0x7FFFFFFFFFFFFFFF**
Fix the bug:
```
PS C:\> Get-DbaFile -SqlInstance test -Path \\dc1\backup -FileType bak

Filename                                        SqlInstance
--------                                        -----------
\dc1\backup\AdventureWorks2016.bak              node1
\dc1\backup\AdventureWorks2016_202401072206.bak node1
\dc1\backup\db1_202401072154.bak                node1

PS C:\>
```

All SMB share paths should start with `\\`.

## Please read -- recent changes to our repo
On November 10, 2022, [we removed some bloat from our repository (for the second and final time)](https://github.com/dataplat/dbatools/issues/8542). This change requires that all contributors reclone or refork their repo.

PRs from repos that have not been recently reforked or recloned will be closed and @potatoqualitee will cherry-pick your commits and open a new PR with your changes.

 - [ ] Please confirm you have the smaller repo (85MB .git directory vs 275MB or 110MB or 185MB .git directory)

## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (affects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/dataplat/appveyor-lab ?
 - [ ] Unit test is included
 - [ ] Documentation
 - [ ] Build system

### CODE REVIEW

File: public/Get-DbaFile.ps1 [hunk 1]
```diff
@@ -191,7 +191,14 @@ function Get-DbaFile {
...
                         if ($row.filename.ToLowerInvariant().EndsWith(".$type")) {
                             $fullpath = $row.fullpath.Replace("\", $separator)
+                            
+                            # Replacing all instances of '\\' with single backslashes '\', and maintain the leading SMB share path represented by the initial '\\'.
+                            $is_smb_share_path = $fullpath.SubString(0, 2) -eq "\\"
                             $fullpath = $fullpath.Replace("\\", "\")
+                            if ($is_smb_share_path) {
+                                $fullpath = $fullpath -replace "^\\", "\\"
+                            }
+
```
Comment (wsmelton) [inline public/Get-DbaFile.ps1:201]:
```powershell
$fullPath = if ($fullpath.Substring(0,2) -eq "\\") {
    # Preserve first two values for UNC path
    $fullpath.Substring(2).Replace("\\","\")
} else {
    $fullPath.Replace("\\","\")
}
```

Comment (wsmelton) [inline public/Get-DbaFile.ps1:201]: I am trying to remember why even try to change the path separator from what SQL returns 🤔

Comment (0x7FFFFFFFFFFFFFFF) [reply public/Get-DbaFile.ps1:201]: In SQL Server, you can create a database like this:

```

CREATE DATABASE [test]
ON PRIMARY
(
    NAME = N'test',
    FILENAME = N'C:\\Program Files\\Microsoft SQL Server\\MSSQL16.MSSQLSERVER\\MSSQL\\DATA\\test.mdf',
    SIZE = 8192KB,
    FILEGROWTH = 65536KB
)
LOG ON
(
    NAME = N'test_log',
    FILENAME = N'C:\\Program Files\\Microsoft SQL Server\\MSSQL16.MSSQLSERVER\\MSSQL\\DATA\\test_log.ldf',
    SIZE = 8192KB,
    FILEGROWTH = 65536KB
);
```

After creating the database, if you query the sys.database_files DMV, you will get a physical_name like `C:\\Program Files\\Microsoft SQL Server\\MSSQL16.MSSQLSERVER\\MSSQL\\DATA\\test.mdf`. Also, if there are flaws in your script when moving database files, you could encounter the same issues. It seems that the replace operation is intended to address these issues.

**Comment by wsmelton** [comment]
I'm not sure why the Action is stuck failing after the base branch is corrected in the PR.

**Comment by potatoqualitee** [comment]
@wsmelton -- maybe its testing the SHA of the specific commit? Either way, I will merge it once appveyor passes bc I can't get it either.

**Pull request merged**